logParser.js

```diff
@@ -25,7 +26,13 @@ if (require.main === module) {
     setInterval(() => {
         db.upkeep();
         db.getWatches((results) => {
-            itemList.add(results);
+            //I think it might be best to just have itemList = results here for a couple of reasons:
+            //a) I like the idea of a set to eleminate searching duplicate data, but I'm pretty sure objects with the same properties do not deeply equate since they have separate memory references (confirmed - just got a 100 pings for 1 watch)
+            //b) it's impossible to have duplicates anyway, as each result object has a username
+            //c) watches expire after 7 days and users can update them at any time - if we just continue adding to itemList, users may get alerts for out of date info.
+
+            //results.forEach((result) => itemList.add(result)) //I refactored your code a bit here to see if it would work (results is an array so need to iterate over it to add to a set), but issue a above just makes a continuing growing list
+            itemList = results;
```

Comment:
Yeah ok, that's fair, I just kinda sprinted through this section since I couldn't run/test it without a test DB, so I didn't think to hard about what that list was. So, the problem then is that this changes the value of the pointer for itemList instead of just mutating the actual data -- which means the pointer that was passed to parseLog is invalid. To make this work, itemList needs to be defined ourside of the scope of this `if` block, I THINK? Unless JS doesn't treat `if` blocks as their own scope, which is definitely possible....

Reply:
This is the way I had it previously and never got any bugs.  Isn't itemList defined much further above already?

Reply:
No, I'd moved it into this if-block -- it's only defined a few lines up, inside this scope. But, if this works, then maybe it's fine (and if-blocks in JS don't have their own scope).